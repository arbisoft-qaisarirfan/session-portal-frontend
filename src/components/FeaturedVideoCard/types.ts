export type FeaturedVideoCardProps = {
  isVisible?: boolean;
  className?: string;
  event_time: string;
  thumbnail?: string;
  title: string;
  description: string;
  video_duration: number;
  workstream_id: string;
  width?: string;
};
