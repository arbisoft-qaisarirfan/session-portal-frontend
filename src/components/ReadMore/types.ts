export interface ReadMoreProps {
  amountOfWords?: number;
  className?: string;
  showLessText: string;
  showMoreText: string;
  text: string;
}
