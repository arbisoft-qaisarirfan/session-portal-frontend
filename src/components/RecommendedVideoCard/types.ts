export interface RecommendedVideoCardProps {
  className?: string;
  date: string;
  duration: string;
  imgUrl: string;
  onClick?: VoidFunction;
  organizer?: string;
  title: string;
  width?: string;
}
