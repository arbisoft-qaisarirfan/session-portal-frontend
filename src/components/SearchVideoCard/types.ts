export type SearchVideoCardProps = {
  id: number;
  className?: string;
  event_time: string;
  thumbnail?: string;
  title: string;
  description: string;
  video_duration: number;
  workstream_id: string;
  width?: string;
};
