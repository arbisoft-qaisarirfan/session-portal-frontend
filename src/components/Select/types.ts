import { SelectProps as MuiSelectProps, SelectChangeEvent } from "@mui/material/Select";

export type SelectProps = {
  menuItems: { value: string; label: string }[];
  handleChange: (event: SelectChangeEvent) => void;
} & MuiSelectProps;
