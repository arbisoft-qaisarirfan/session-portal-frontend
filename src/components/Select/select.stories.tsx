import { Meta, StoryFn } from "@storybook/react";

import Select from "./select";
import { SelectProps } from "./types";

export default {
  title: "Components/Select",
  component: Select,
  tags: ["autodocs"],
  parameters: { jest: "select.test.tsx" },
} as Meta;

export const Default: StoryFn<SelectProps> = (args) => <Select {...args} />;
Default.args = {
  label: "Sort By",
  menuItems: [
    { value: "Newest", label: "Newest" },
    { value: "Oldest", label: "Oldest" },
  ],
  value: "",
};

export const DisabledState: StoryFn<SelectProps> = (args) => <Select {...args} />;
DisabledState.args = {
  label: "Sort By",
  menuItems: [
    { value: "Newest", label: "Newest" },
    { value: "Oldest", label: "Oldest" },
  ],
  value: "",
  disabled: true,
};

export const CustomValue: StoryFn<SelectProps> = (args) => <Select {...args} />;
CustomValue.args = {
  label: "Sort By",
  menuItems: [
    { value: "Newest", label: "Newest" },
    { value: "Oldest", label: "Oldest" },
    { value: "Most Popular", label: "Most Popular" },
  ],
  value: "Newest",
};
