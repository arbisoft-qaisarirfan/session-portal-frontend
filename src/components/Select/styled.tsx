import Box from "@mui/material/Box";
import { selectClasses } from "@mui/material/Select";
import { styled, css, alpha } from "@mui/material/styles";

export const SelectWrapper = styled(Box, {
  name: "SelectWrapper",
})(
  ({ theme }) => css`
    align-items: center;
    display: flex;
    gap: 12px;
    min-width: 124px;

    .${selectClasses.select} {
      background-color: ${alpha(theme.palette.colors.white, 0.2)};
      font-size: 14px;
      font-weight: 600;
    }
  `
);
