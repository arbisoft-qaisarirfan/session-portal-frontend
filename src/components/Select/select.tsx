import React from "react";

import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import MuiSelect from "@mui/material/Select";

import { SelectWrapper } from "./styled";
import { SelectProps } from "./types";

const Select: React.FC<SelectProps> = ({ onChange, value, menuItems, labelId, label }) => {
  return (
    <SelectWrapper>
      <InputLabel>{label}</InputLabel>
      <MuiSelect value={value} onChange={onChange} displayEmpty data-testid="sort-select" labelId={labelId} size="small">
        {menuItems.map((item) => (
          <MenuItem data-testid={item.value} key={item.value} value={item.value}>
            {item.label}
          </MenuItem>
        ))}
      </MuiSelect>
    </SelectWrapper>
  );
};

export default Select;
