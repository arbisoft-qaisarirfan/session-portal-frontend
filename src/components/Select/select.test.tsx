import React from "react";

import { act, customRender, fireEvent, screen } from "@/jest/utils/testUtils";

import Select from "./select";
import { SelectProps } from "./types";

describe("Select Component", () => {
  const mockHandleChange = jest.fn();

  const setup = (props?: Partial<SelectProps>) => {
    const defaultProps: SelectProps = {
      menuItems: ["Option 1", "Option 2", "Option 3"],
      handleChange: mockHandleChange,
      value: "",
    };

    return customRender(<Select {...defaultProps} {...props} />);
  };

  it("should render the select component with default option", () => {
    setup();
    expect(screen.getByText("Sort By")).toBeInTheDocument();
  });

  it("should render all menu items", async () => {
    const { getByRole, getByTestId } = setup();
    fireEvent.mouseDown(getByRole("combobox"));
    await act(async () => {
      expect(getByTestId("Option 1")).toBeInTheDocument();
      expect(getByTestId("Option 2")).toBeInTheDocument();
      expect(getByTestId("Option 3")).toBeInTheDocument();
    });
  });

  it("should call handleChange when an option is selected", async () => {
    const { getByTestId, getByRole } = setup();
    fireEvent.mouseDown(getByRole("combobox"));
    await act(async () => {
      getByTestId("Option 1").click();
    });
    expect(mockHandleChange).toHaveBeenCalledTimes(1);
  });

  it("should display the selected value correctly", () => {
    const { getByRole } = setup({ value: "Option 3" });
    expect(getByRole("combobox")).toHaveTextContent("Option 3");
  });

  it("should not crash if menuItems is empty", () => {
    const { getByRole } = setup({ menuItems: [] });
    fireEvent.mouseDown(getByRole("combobox"));
    expect(screen.queryByText("Option 1")).not.toBeInTheDocument();
  });

  test("should match snapshot", () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });
});
