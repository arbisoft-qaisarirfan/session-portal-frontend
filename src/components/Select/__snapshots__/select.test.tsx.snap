// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root css-6bv9qa-MuiFormControl-root-SelectFormControl-root"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-8r70ge-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      data-testid="sort-select"
    >
      <div
        aria-controls=":r5:"
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-1hhzsab-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        role="combobox"
        tabindex="0"
      >
        Sort By
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1exavk5-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1v24f9t-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-w4cd9x"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
